### ISSUE TITLE
ActiveDirectory authentication doesn't work on Linux

### ISSUE THREAD

**Issue by Richard-Payne** [microsoft/azuredatastudio#8713]
- Versions:
Version: 1.13.1
Commit: d08d07fe39c7fe74f9265db02cb963c38d048299
Date: 2019-11-14T22:56:48.883Z
VS Code: 1.38.0
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Linux x64 4.15.0-72-generic

Steps to Reproduce:

1. Open Azure Data Studio
2. Add Connection
3. Type <servername>\sqlexpress
4. Set Authentication Type to "Windows Authentication" 
5. Click Connect
6. Receive message:
     Connection failed due to Kerberos error.
     Help configuring Kerberos is available at https://aka.ms/sqlopskerberos
     If you have previously connected you may need to re-run kinit.
7. click "Run kinit"
8. enter password in the terminal that opens.
9. Repeat from step 2.

My machine is domain joined using realmd and I can successfully login to Linux using my domain user.  kinit succeeds and klist shows the valid ticket.

Windows Authentication works fine from a Windows machine.

**Comment by Richard-Payne** [comment]
@Charles-Gagnon  any response to this?

**Comment by Charles-Gagnon** [comment]
If you open the developer tools (Help -> Toggle Developer tools) and look at the console tab is there any errors displayed there?

Are you running klist from within an ADS terminal window? If not could you try that and see if you still see the valid ticket?

**Comment by Richard-Payne** [comment]
This appears in the dev tools console whenever I click the connect button:
```
 INFO Creating new connection connection:providerName:MSSQL|applicationName:azdata|authenticationType:Integrated|database:|server:ldbcfg001\sqlexpress|user:|group:C777F06B-202E-4480-B475-FA416154D458
/usr/share/azuredatastudio/resources/app/node_modules.asar/zone.js/dist/zone-node.js:2280  INFO Adding connection connection:providerName:MSSQL|applicationName:azdata|authenticationType:Integrated|database:|server:ldbcfg001\sqlexpress|user:|group:C777F06B-202E-4480-B475-FA416154D458
/usr/share/azuredatastudio/resources/app/node_modules.asar/zone.js/dist/zone-node.js:2280  INFO Successfully added connection connection:providerName:MSSQL|applicationName:azdata|authenticationType:Integrated|database:|server:ldbcfg001\sqlexpress|user:|group:C777F06B-202E-4480-B475-FA416154D458
/usr/share/azuredatastudio/resources/app/node_modules.asar/zone.js/dist/zone-node.js:2280  INFO Deleting connection connection:providerName:MSSQL|applicationName:azdata|authenticationType:Integrated|database:|server:ldbcfg001\sqlexpress|user:|group:C777F06B-202E-4480-B475-FA416154D458
```
I have been able to connect to a different server using Windows Authentication.  So, what I have is:

Sql Management Studio  on Windows => Server A  (works)
Sql Management Studio  on Windows => Server B  (works)
Azure Data Studio on Linux => Server A  (works)
Azure Data Studio on Linux => Server B  (fails)

There is a kerberos ticket created when I login to linux using my domain user.  This viewable in klist from both a normal terminal and an ADS embedded terminal.

After successfully connecting to Server A, I now have two tickets, the original login ticket and one specific to Server A.

The only potential difference I can see is that Server B is using an instance.  Eg:  

Server A address:  servera
Server B address: serverb\sqlexpress

**Comment by Richard-Payne** [comment]
@Charles-Gagnon is there any schedule for when this will be fixed?

**Comment by Charles-Gagnon** [comment]
@Richard-Payne Could you try out the new sign in experience by following the instructions [here](https://github.com/microsoft/azuredatastudio/issues/7619#issuecomment-566231484) to see if that solves your issue?

**Comment by Richard-Payne** [comment]
So I ran:
`NEW_SIGN_IN_EXPERIENCE=true /usr/share/azuredatastudio/azuredatastudio --unity-launch`

It didn't fix the problem, but then, the "experience" didn't seem any different either.  Should I be expected a new/different frontend workflow?

**Comment by Charles-Gagnon** [comment]
Have you tried adding the server port after the server name? That seems to have helped some people per https://github.com/microsoft/azuredatastudio/issues/447#issuecomment-546407390

**Comment by Richard-Payne** [comment]
Adding the port did not help.

**Comment by festerman** [comment]
The port number did work for me ... 

Version: 1.14.1
Commit: f9b968c1ae514a0d1210788629e1c69125857279
Date: 2019-12-24T02:35:43.481Z
VS Code: 1.38.0
Electron: 6.1.5
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Linux x64 5.4.8-arch1-1

**Comment by Richard-Payne** [comment]
Any further ideas @Charles-Gagnon?

**Comment by Charles-Gagnon** [comment]
Sorry, nothing for fixing the specific issue you're having. Although would using docker be an option? That might get around it if the issue is named instances are causing an issue here.

 There's been a lot of reports of various issues around AAD so we're working on investigating those but they've been difficult to track down due to the huge variety in network setups people have. I'll let you know if we need any further information or have any other suggestions for you to try.

**Comment by Richard-Payne** [comment]
@Charles-Gagnon Docker is not a problem.  However, I see no official packages for Azure Data Studio.  I tried both community offerings, neither of which launched correctly.  What are you suggesting?

**Comment by Charles-Gagnon** [comment]
I meant making a docker container running your SQLEXPRESS instance and connecting to that. If for some reason the named instance is causing a problem here then having it be the default instance in the docker container might be a workaround. 

Or even spinning up a VM and hosting the instance there instead.

**Comment by Richard-Payne** [comment]
That's not really practical in this case.

**Comment by cheenamalhotra** [comment]
Hi @Richard-Payne 

Would request you to try and confirm if the issue is fixed with **v1.39** onwards?
Appreciate your time!

**Comment by cheenamalhotra** [comment]
I will proceed to close this issue as Kerberos related issues on Linux are expected to be fixed with v1.39 onwards.
Please open a new issue for assistance if needed.
